modules/apps/27-interchain-accounts/host/ibc_module.go

```diff
@@ -111,7 +111,16 @@ func (im IBCModule) OnRecvPacket(
 	ack := channeltypes.NewResultAcknowledgement([]byte{byte(1)})
 
 	if err := im.keeper.OnRecvPacket(ctx, packet); err != nil {
-		ack = channeltypes.NewErrorAcknowledgement(err.Error())
+		ack = channeltypes.NewErrorAcknowledgement("error occurred handling packet on destination chain")
+
+		ctx.EventManager().EmitEvent(
```

Comment:
lets make a function for this and add it into `keeper/events.go` - #648

Reply:
Agreed, prefer this approach. Updated